### ISSUE TITLE
Dashboard is not showing after upgrade to 1.8.5

### ISSUE THREAD

**Issue by OrgesKreka** [HangfireIO/Hangfire#2307]
Lately I upgraded to Hangfire 1.8.5 but there is an issue with the dashboard UI, as I can see the jobs are executed.
This is the error that is shown:
![image](https://github.com/HangfireIO/Hangfire/assets/22299171/2ba0ae4d-d8da-4b1f-8a6e-5d4ee5e9e54f)
This are the packages that I am using:
```xml
<ItemGroup>
    <PackageReference Include="Hangfire.AspNetCore" Version="1.8.5" />
    <PackageReference Include="Hangfire.Console" Version="1.4.2" />
    <PackageReference Include="Hangfire.Core" Version="1.8.5" />
    <PackageReference Include="Hangfire.SqlServer" Version="1.8.5" />
    <PackageReference Include="Microsoft.Data.SqlClient" Version="5.1.1" />
  </ItemGroup>
```
And this is the configuration part:

```csharp
builder.Services.AddHangfire(config => config
       .SetDataCompatibilityLevel(CompatibilityLevel.Version_170)
       .UseSimpleAssemblyNameTypeSerializer()
       .UseRecommendedSerializerSettings()
       .UseConsole()
       .UseSqlServerStorage(() => new SqlConnection(connectionString), new SqlServerStorageOptions
       {
           //SqlClientFactory = Microsoft.Data.SqlClient.SqlClientFactory, // --> This is not working!
           CommandTimeout = TimeSpan.FromMinutes(5),
           SchemaName = "Hangfire",
           SlidingInvisibilityTimeout = TimeSpan.FromMinutes(5),
           QueuePollInterval = TimeSpan.Zero,
           UseRecommendedIsolationLevel = true,
           DisableGlobalLocks = true,
           PrepareSchemaIfNecessary = true
       }));

builder.Services.AddHangfireServer();
builder.Services.AddControllers();

var app = builder.Build();

// Configure the HTTP request pipeline.
app.UseHttpsRedirection();

app.UseRouting();
app.UseHangfireDashboard();

app.UseEndpoints(endpoints =>
{
    endpoints.MapHangfireDashboard(new DashboardOptions
    {
        DarkModeEnabled = true
    });
});
```
Someone can help me how to fix this issue?

**Comment by odinserj** [comment]
It looks like for some reason, your `HangFire.Schema` table doesn't contain a required row that describes the current schema version. To make Dashboard UI work, please insert a value `7` there. Please note that this is just an educated guess, and we still need to understand what's your current schema version. Let's use the following algorithm for determining it:

* **Version 9**: if there's the `IX_HangFire_State_CreatedAt` index on the `State` table;
* **Version 8**: if there's the `PK_HangFire_Counter` primary key on the `Counter` table based on both `Key` and `Id` columns;
* **Version 7**: if there's the `IX_HangFire_Set_Score` index on the `Set` table based on both `Key` and `Score` columns;
* **Version 6**: if there's the `PK_HangFire_State` primary key on the `State` table based on both `JobId` and `Id` columns;
* **Version 5**: if `Server.Id` column has `NVARCHAR (200)` type;
* **Version 4**: if `AggregatedCounter` table exists;
* **Version 3**: if `UX_HangFire_Hash_Key_Field` index exists for table `Hash`'
* **Version 2**: if `Value` table does not exist;
* **Version 1**: if `Value` table exists.

You can check conditions from top to down and stop when first match is encountered.

I will also fix this in 1.8.6 to give alert on the corresponding "Schema version" metric instead of throwing an exception.

**Comment by OrgesKreka** [comment]
The hack by adding version 7 to Schema table works, also the upgrade to v1.8.6 works in this case the version is shown as 'Unspecified Schema version'.

Thanks a lot!
